model/setproperty.py

```diff
@@ -65,21 +63,39 @@ def validate_set_contents(self, value):
       BadValueError if the set has items are not instances of the
       item_type given to the constructor.
     """
-    if self.item_type in (int, long):
-      item_type = (int, long)
-    else:
-      item_type = self.item_type
 
     for item in value:
-      if not isinstance(item, item_type):
-        if item_type == (int, long):
-          raise db.BadValueError('Items in the %s set must all be integers.' %
-                              self.name)
-        else:
-          raise db.BadValueError(
-              'Items in the %s set must all be %s instances' %
-              (self.name, self.item_type.__name__))
+      if not isinstance(item, self.item_type):
+        raise db.BadValueError(
+            'Items in the %s set must all be %s instances' %
+            (self.name, self.item_type.__name__))
     return value
+  
+  # def validate_set_contents(self, value):
```

Comment:
You can probably delete this commented chunk safely - looks like it was written for Python 2.